Source/Initialization/WarpXInitData.cpp

```diff
@@ -231,23 +233,189 @@ WarpX::PostRestart ()
     mypc->PostRestart();
 }
 
+
+namespace {
+WarpXParser makeParser (std::string const& parse_function)
+{
+    WarpXParser parser(parse_function);
+    parser.registerVariables({"x","y","z"});
+    ParmParse pp("my_constants");
+    std::set<std::string> symbols = parser.symbols();
+    symbols.erase("x");
+    symbols.erase("y");
+    symbols.erase("z");
+    for (auto it = symbols.begin(); it != symbols.end(); ) {
+        Real v;
+        if (pp.query(it->c_str(), v)) {
+           parser.setConstant(*it, v);
+           it = symbols.erase(it);
+        } else {
+           ++it;
+        }
+    }
+    for (auto const& s : symbols) {
+        amrex::Abort(" ExternalEBFieldOnGrid::makeParser::Unknown symbol "+s);
+    }
+    return parser;
+}
+}
+
+
+/* \brief
```

Comment:
Thanks for this description! Could you put it in the `.H` file at the function declaration instead, and format it 
```c++
/**
 * \brief
``` 
instead?

Reply:
Done. Thank you for the suggestion